Docs number text date component (#1892)
Altinn/altinn-studio-docs: content/altinn-studio/reference/ux/components/Text/_index.en.md

---
title: Text
linktitle: Text
description: A component that allows you to display text with or without label.
schemaname: Text # Component schema name used to autogenerate list of properties from json schema (replace with appropriate component name)
weight: 10 # Do not change, the components will be sorted alphabetically
toc: true
---

## Usage

> The Text component allows users to add structured text, with or without label

Reviewer: ```suggestion
The `Text` component allows users to add structured text, with or without label.
```

|------------------------------|----------|------------------------------------------------------------------------------------------|
| `id`                         | string   | Unique Id string for the component                                                       |
| `value`                      | string   | The text you want to display.                                                            |
| `textResourceBindings.title` | string   | Label of the text you want to display                                                    |
| `direction`                  | string   | Sets the structured direction of label and value. **Enum:** `"horizontal" \| "vertical"` |
| `icon`                       | string   | An URL path to the image/icon                                                            |

## Configuration

### Add component

```json{hl_lines="6-"}
{
  "id": "text",
  "type": "Text",
  "textResourceBindings": {
    "title": "Name"
  },
  "value": "My name",
  "direction": "horizontal",
  "icon": "https://www.example.com/icon.png"
},
```

### Add title and elements

<br>

#### `textResourceBindings.title`

> The title for the text is shown as a label and can be added as text directly or referenced via a text key to a [text resource](/nb/altinn-studio/reference/ux/texts/#legge-til-og-endre-tekster-i-en-app).

Reviewer: ```suggestion
Label for the date. This can be added as a hardcoded string or as a reference to a [text resource](/nb/altinn-studio/reference/ux/texts/#legge-til-og-endre-tekster-i-en-app).
```

#### `value`

Displays the value for the label and can be added as a string or an expression.

#### `direction`

> Decides if the text should is shown under or next to the label. horizontal for "next to" and vertical for "under". 

Reviewer: ```suggestion
Decides if the number should be shown under or next to the label. `horizontal` for showing it next to the label and `vertical` for under. 
```

### Behavior

(How the component behaves in different contexts)

### Style

(Visual styling (e.g. alignment, padding, dos and don'ts))

### Best Practices

(Industry standards, dos and don'ts)

### Content guidelines

(E.g. punctuation rules, standard labels, etc.)

### Accessibility

(Component-specific best practices for accessibility.)

### Mobile

(How to apply component in mobile environments.)

-->
### Related

- [`Number`](../number/)
- [`Date`](../date/)

## Properties

{{% notice warning %}}
We are currently updating how we implement components, and the list of properties may not be entirely accurate.
{{% /notice %}}

| **Property**                 | **Type** | **Description**                                                                          |
|------------------------------|----------|------------------------------------------------------------------------------------------|
| `id`                         | string   | Unique Id string for the component                                                       |
| `value`                      | string   | The text you want to display.                                                            |

> | `textResourceBindings.title` | string   | Label of the text you want to display                                                    |
> | `direction`                  | string   | Sets the structured direction of label and value. **Enum:** `"horizontal" \| "vertical"` |
> | `icon`                       | string   | An URL path to the image/icon                                                            |

Reviewer: ```suggestion
| `textResourceBindings.title` | string   | Label of the text you want to display.                                                    |
| `direction`                  | string   | Sets the structured direction of label and value. <br/><br/>**Enum:** [horizontal, vertical]<br/><br/>**Default:** horizontal |
| `icon`                       | string   | An URL path to the image/icon.                                                            |
```

> | `icon`                       | string   | An URL path to the image/icon                                                            |

Reviewer: Fjernes ref kommentar over (ikon)?

## Configuration

### Add component

```json{hl_lines="6-"}
{
  "id": "text",
  "type": "Text",
  "textResourceBindings": {
    "title": "Name"
  },
  "value": "My name",
  "direction": "horizontal",
  "icon": "https://www.example.com/icon.png"
},
```

### Add title and elements

<br>

#### `textResourceBindings.title`

Label for the date. This can be added as a hardcoded string or as a reference to a [text resource](/nb/altinn-studio/reference/ux/texts/#legge-til-og-endre-tekster-i-en-app).

#### `value`

The value of the text to be displayed. Can be added as a string or an expression.

#### `direction`

Decides if the number should be shown under or next to the label. `horizontal` for showing it next to the label and `vertical` for under. 

<!-- 
#### `icon`

If you want to show an icon before the label, you can add an URL path.
-->

<br>

> #### Example

Reviewer: Fjernes ref kommentar over?
